SmartHome/SmartHome/src/main/java/smarthome/Host.java

```diff
@@ -0,0 +1,104 @@
+package smarthome;
+/* This class implements the smart home -host behaviour
+* The devices of the smart home are hardcoded into the dev[] array
+* Every method cast the instance of the array to the specific subclass
+*
+* */
+class Host{
+    private Devices[] dev = new Devices[10];
+
+    Host() {
+        dev[0] = new SmartThermostat();
+        dev[1] = new SmartLamp();
+        dev[2] = new SmartCamera();
+    }
+
+     private void getThermostatStatus() {
```

Comment:
As mentioned during class, this approach of having a method for the features of each smart device does not scale elegantly. You are breaking the Open Closed and Single Responsibility OOP principles.

It makes more sense to obtain a collection of all smart devices from the Host, and then interact directly with them.